pkg/watch/until.go

```diff
@@ -23,7 +23,9 @@ import (
 )
 
 // ConditionFunc returns true if the condition has been reached, false if it has not been reached yet,
-// or an error if the condition cannot be checked and should terminate.
+// or an error if the condition cannot be checked and should terminate. In general, it is better to define
+// level driven conditions over edge driven conditions (pod has ready=true, vs pod modified and ready changed
+// from false to true).
```

Comment:
I would come out and just say that if you test for edges, you'll probably end up with something that doesn't work reliably unless you're super careful, but that's just me :)